### PR TITLE
En_Sth OK

### PR DESCRIPTION
**Pull request by gamestabled**
made a c file for the 700 lines of gfx data

### CODE REVIEW

File: src/overlays/actors/ovl_En_Sth/z_en_sth.c [hunk 1]
```diff
@@ -40,45 +48,357 @@ static ColliderCylinderInit D_80B0B404 = {
...
+void EnSth_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnSth* this = THIS;
+
+    s16 objectId;
+    s32 params = this->actor.params;
+    s32 objectBankIdx;
+    s32 params2;
+
+    // Translation: Gold Skulltula Shop
+    osSyncPrintf("\x1b[34m金スタル屋 no = %d\n\x1b[m", params);
```
Comment (Random06457) [inline src/overlays/actors/ovl_En_Sth/z_en_sth.c:96]:
```suggestion
    osSyncPrintf(VT_FGCOL(BLUE) "金スタル屋 no = %d\n" VT_RST, params);
```
Note that you can use `./tools/vt_fmt.py "string to format"` to format strings properly.

File: src/overlays/actors/ovl_En_Sth/z_en_sth.c [hunk 2]
```diff
@@ -40,45 +48,357 @@ static ColliderCylinderInit D_80B0B404 = {
...
+        if (gSaveContext.inventory.gsTokens < 100) {
+            Actor_Kill(&this->actor);
+            // Translation: Gold Skulltula Shop I still can't be a human
+            osSyncPrintf("金スタル屋 まだ 人間に戻れない \n");
+            return;
+        }
+    } else if (gSaveContext.inventory.gsTokens < (params2 * 10)) {
+        Actor_Kill(&this->actor);
+        // Translation: Gold Skulltula Shop I still can't be a human
+        osSyncPrintf("\x1b[34m金スタル屋 まだ 人間に戻れない \n\x1b[m");
```
Comment (Random06457) [inline src/overlays/actors/ovl_En_Sth/z_en_sth.c:108]:
```suggestion
        osSyncPrintf(VT_FGCOL(BLUE) "金スタル屋 まだ 人間に戻れない \n" VT_RST);
```

File: src/overlays/actors/ovl_En_Sth/z_en_sth.c [hunk 3]
```diff
@@ -40,45 +55,357 @@ static ColliderCylinderInit D_80B0B404 = {
...
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sth/func_80B07540.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sth/EnSth_Init.s")
+static s16 sObjectIds[6] = {
+    OBJECT_AHG, OBJECT_BOJ, OBJECT_BOJ, OBJECT_BOJ, OBJECT_BOJ, OBJECT_BOJ,
+};
+
+static FlexSkeletonHeader* sSkeletons[6] = {
+    0x060000F0, 0x060000F0, 0x060000F0, 0x060000F0, 0x060000F0, 0x060000F0,
+};
```
Comment (petrie911) [inline src/overlays/actors/ovl_En_Sth/z_en_sth.c:65]: lol, such a necessary array. Though shouldn't these be replaced with their proper symbols.

File: src/overlays/actors/ovl_En_Sth/z_en_sth.c [hunk 4]
```diff
@@ -40,45 +55,357 @@ static ColliderCylinderInit D_80B0B404 = {
...
+    this->actionFunc = actionFunc;
+}
+
+void EnSth_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnSth* this = THIS;
+
+    s16 objectId;
+    s32 params = this->actor.params;
+    s32 objectBankIdx;
+    s32 params2;
```
Comment (petrie911) [inline src/overlays/actors/ovl_En_Sth/z_en_sth.c:100]: are these temps really needed. Could this be fixed using `thisx`?

File: src/overlays/actors/ovl_En_Sth/z_en_sth.c [hunk 5]
```diff
@@ -40,45 +55,357 @@ static ColliderCylinderInit D_80B0B404 = {
...
+
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 36.0f);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    this->actor.colChkInfo.mass = MASS_IMMOVABLE;
+    this->actor.update = EnSth_Update2;
+    this->actor.draw = this->drawFunc;
+}
+
+void EnSth_SetupAnimation(EnSth* this, GlobalContext* globalCtx) {
```
Comment (petrie911) [inline src/overlays/actors/ovl_En_Sth/z_en_sth.c:150]: These seem to do more than just setup the collider and animations. Perhaps the names could be improved.

File: src/overlays/actors/ovl_En_Sth/z_en_sth.c [hunk 6]
```diff
@@ -40,45 +55,357 @@ static ColliderCylinderInit D_80B0B404 = {
...
+    s32 pad;
+    s16* params;
```
Comment (petrie911) [inline src/overlays/actors/ovl_En_Sth/z_en_sth.c:152]: You sure about this one?

File: src/overlays/actors/ovl_En_Sth/z_en_sth.c [hunk 7]
```diff
@@ -40,45 +55,357 @@ static ColliderCylinderInit D_80B0B404 = {
...
+        this->actionFunc = EnSth_SetupAnimation;
+    }
+}
+
+void EnSth_FacePlayer(EnSth* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s16 diffRot;
+
+    diffRot = this->actor.yawTowardsPlayer - this->actor.shape.rot.y;
+    if (ABS(diffRot) < 0x4001) {
```
Comment (petrie911) [inline src/overlays/actors/ovl_En_Sth/z_en_sth.c:187]:
```suggestion
    if (ABS(diffRot) <= 0x4000) {
```

File: src/overlays/actors/ovl_En_Sth/z_en_sth.c [hunk 8]
```diff
@@ -40,45 +55,357 @@ static ColliderCylinderInit D_80B0B404 = {
...
+        this->actor.objBankIndex = this->objectBankIdx;
+        this->actionFunc = EnSth_SetupAnimation;
+    }
+}
+
+void EnSth_FacePlayer(EnSth* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s16 diffRot;
+
+    diffRot = this->actor.yawTowardsPlayer - this->actor.shape.rot.y;
```
Comment (petrie911) [inline src/overlays/actors/ovl_En_Sth/z_en_sth.c:186]:
```suggestion
    s16 diffRot = this->actor.yawTowardsPlayer - this->actor.shape.rot.y;
    
```

File: src/overlays/actors/ovl_En_Sth/z_en_sth.c [hunk 9]
```diff
@@ -40,45 +55,357 @@ static ColliderCylinderInit D_80B0B404 = {
...
+        Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 0xC, 0x3E8, 0x64);
+        this->actor.world.rot.y = this->actor.shape.rot.y;
+    }
+}
+
+void EnSth_LookAtPlayer(EnSth* this, GlobalContext* globalCtx) {
+    s16 diffRot;
+
+    diffRot = this->actor.yawTowardsPlayer - this->actor.shape.rot.y;
+    if ((ABS(diffRot) < 0x4301) && (this->actor.xzDistToPlayer < 100.0f)) {
```
Comment (petrie911) [inline src/overlays/actors/ovl_En_Sth/z_en_sth.c:206]:
```suggestion
    s16 diffRot = this->actor.yawTowardsPlayer - this->actor.shape.rot.y;
    
    if ((ABS(diffRot) <= 0x4300) && (this->actor.xzDistToPlayer < 100.0f)) {
```

File: src/overlays/actors/ovl_En_Sth/z_en_sth.c [hunk 10]
```diff
@@ -40,45 +55,357 @@ static ColliderCylinderInit D_80B0B404 = {
...
+
+void EnSth_RewardUnobtainedWait(EnSth* this, GlobalContext* globalCtx) {
+    if (func_8002F194(&this->actor, globalCtx)) {
+        EnSth_SetupAction(this, EnSth_RewardUnobtainedTalk);
+    } else {
+        if (this->actor.params == 0) {
+            this->actor.textId = 0x28; // "Since you've destroyed [gsTokens] Spiders ..."
+        } else {
+            this->actor.textId = 0x21; // "The curse has been broken! ..."
+        }
```
Comment (petrie911) [inline src/overlays/actors/ovl_En_Sth/z_en_sth.c:288]: possible ternary? I suppose that makes the comments not really work.

File: src/overlays/actors/ovl_En_Sth/z_en_sth.c [hunk 11]
```diff
@@ -40,45 +55,357 @@ static ColliderCylinderInit D_80B0B404 = {
...
+
+void EnSth_Update2(Actor* thisx, GlobalContext* globalCtx) {
+    EnSth* this = THIS;
+    s32 pad;
+
+    Collider_UpdateCylinder(&this->actor, &this->collider);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    Actor_MoveForward(&this->actor);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 0.0f, 0.0f, 0.0f, 4);
+    if (SkelAnime_Update(&this->skelAnime) != 0) {
```
Comment (petrie911) [inline src/overlays/actors/ovl_En_Sth/z_en_sth.c:326]:
```suggestion
    if (SkelAnime_Update(&this->skelAnime)) {
```

File: src/overlays/actors/ovl_En_Sth/z_en_sth.c [hunk 12]
```diff
@@ -40,45 +55,357 @@ static ColliderCylinderInit D_80B0B404 = {
...
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sth/func_80B07878.s")
+    if (limbIndex == 15) {
+        rot->x += this->headRot.y;
+        rot->z += this->headRot.x;
+        *dList = D_80B0A050;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sth/func_80B078C8.s")
+    if ((this->unk_2B2 & 2) != 0) {
```
Comment (petrie911) [inline src/overlays/actors/ovl_En_Sth/z_en_sth.c:351]:
```suggestion
    if (this->unk_2B2 & 2) {
```

**Review by petrie911 — COMMENTED** [review]
Kind of a weird actor, especially with the update stuff.

File: src/overlays/actors/ovl_En_Sth/z_en_sth.c (hunk 3 above)
Comment (gamestabled) [reply src/overlays/actors/ovl_En_Sth/z_en_sth.c:65]: these objects files for these arent "decompiled" yet, ill probably do them soon and theyll either make it into this PR before it gets merged, or if not they should make it into my "various objects" PR

File: src/overlays/actors/ovl_En_Sth/z_en_sth.c (hunk 4 above)
Comment (gamestabled) [reply src/overlays/actors/ovl_En_Sth/z_en_sth.c]: it seems like the params going into the osSyncPrintf and into the sObjectIds need to be the same, and the params inside the if/else need to be the same, but these two pairs need to be different. doing it the way i have now only uses 1 temp and avoids thisx

File: src/overlays/actors/ovl_En_Sth/z_en_sth.c (hunk 6 above)
Comment (gamestabled) [reply src/overlays/actors/ovl_En_Sth/z_en_sth.c:150]: nope its definitely weird but its the best i could get that matched, idk

File: src/overlays/actors/ovl_En_Sth/z_en_sth.c (hunk 10 above)
Comment (gamestabled) [reply src/overlays/actors/ovl_En_Sth/z_en_sth.c:284]: luckily the ternary doesnt match anyway :) lol

**Review by Roman971 — APPROVED** [review]

**Comment by fig02** [comment]
gonna remove the text id comments myself in a sec here, but other than that good to go

**Review by fig02 — APPROVED** [review]

**Pull request merged**